apis/cluster/v1beta1/membercluster_types.go

```diff
@@ -55,6 +56,23 @@ type MemberClusterSpec struct {
 	Taints []Taint `json:"taints,omitempty"`
 }
 
+// MetricName is the name of a non-resource metric; it should be a Kubernetes label name.
+type MetricName string
+
+// MetricValue is the value of a non-resource metric;
+type MetricValue struct {
+	// Value is the value of the non-resource metric, which is a Kubernetes quantity.
+	// For more information, see
+	// https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.
+	//
+	// +required
+	Value resource.Quantity `json:"value"`
```

Comment:
What is our plan to support cluster features like:
hasGPU
SupportedVMSizes

Will we have metrics whose values are not quantity but a real string?

It might be more extensible to keep value as a string.

Reply:
Will we add properties in addition to metrics later?

![image](https://github.com/Azure/fleet/assets/13264318/ffec8742-d807-4f96-b59f-f200d8a64f3f)

Reply:
Hi Liqian! It has been updated; as suggested, all refs to metrics have now become properties to be more generic